client/landing/stepper/declarative-flow/internals/steps-repository/site-vertical/form.tsx

```diff
@@ -6,20 +6,22 @@ import FormFieldset from 'calypso/components/forms/form-fieldset';
 import FormSettingExplanation from 'calypso/components/forms/form-setting-explanation';
 import SelectVertical from 'calypso/components/select-vertical';
 import { tip } from 'calypso/signup/icons';
+import type { Vertical } from 'calypso/components/select-vertical/types';
 import './form.scss';
 
 interface Props {
 	isSkipSynonyms?: boolean;
+	onSelect?: ( vertical: Vertical ) => void;
 	onSubmit?: ( event: React.FormEvent ) => void;
 }
 
-const SiteVerticalForm: React.FC< Props > = ( { isSkipSynonyms, onSubmit } ) => {
+const SiteVerticalForm: React.FC< Props > = ( { isSkipSynonyms, onSelect, onSubmit } ) => {
 	const translate = useTranslate();
 
 	return (
 		<form className="site-vertical__form" onSubmit={ onSubmit }>
 			<FormFieldset className="site-vertical__form-fieldset">
-				<SelectVertical isSkipSynonyms={ isSkipSynonyms } />
+				<SelectVertical isSkipSynonyms={ isSkipSynonyms } onSelect={ onSelect } />
```

Comment:
Does SelectVertical support default value? IMO, we should set the current value to indicate the selected vertical when the user goes back to this step.

Reply:
Yes, it will be part of the implementation :-)

Reply:
@arthur791004 on second thought, I will do this in a follow-up PR since I don't want this PR to get too big.